code/Forms/UnsavedChangesIndicator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace SilverStripe\Admin\Forms;
+
+use RuntimeException;
+use SilverStripe\Forms\FormAction;
+
+/**
+ * A visual indicator to show whether a form has unsaved changes.
+ *
+ * Note that this isn't strictly a FormAction as it doesn't perform any action when clicked.
+ * However, extending FormAction allows us to easily integrate with the existing form action
+ * architecture (e.g. placement in the form actions area).
+ */
+class UnsavedChangesIndicator extends FormAction
+{
+    private static array $minutes = [
+        'notice' => 5,
+        'warning' => 10,
+    ];
+
+    protected $schemaComponent = 'UnsavedChangesIndicator';
+
+    public function getSchemaDataDefaults()
+    {
+        // This is used for initialising the React component
+        // Refer to silverstrpie/admin UnsavedChangesIndicator.js
```

Comment:
```suggestion
        // Refer to client/src/components/UnsavedChangesIndicator
```
This is a bit of a nitpick - I won't delay merging if this change isn't made

Reply:
Have updated though without the full path, you can easily just search for the filename